### PR TITLE
Recommendations, upvotes/downvotes, video rating count

### PR DESCRIPTION
**Pull request by horahoradev**
Adds:
- recommendations
- upvotes/downvotes
- rating count integration to video player page

### CODE REVIEW

File: pages/videos/kv[kvid].vue [hunk 1]
```diff
@@ -64,6 +74,19 @@
 					:fans="videoDetails?.userSubscribers ?? 0"
 					isFollowed
 				/>
+				<Subheader id="recheader" icon="upload" :badge="5">Recommendations </Subheader>
```
Comment (horahoradev) [inline pages/videos/kv[kvid].vue]: pay attention here

File: components/Player/PlayerVideo/PlayerVideoPanel/PlayerVideoPanel.vue [hunk 2]
```diff
@@ -16,6 +20,20 @@
 		useEvent("app:requestLogin");
 	}
 
+	/** upvotes the current video */
+	async function Upvote() {
```
Comment (sidd) [inline components/Player/PlayerVideo/PlayerVideoPanel/PlayerVideoPanel.vue]: lowercase function/method names in js, here and `Downvote`

File: components/Player/PlayerVideo/PlayerVideoPanel/PlayerVideoPanel.vue [hunk 3]
```diff
@@ -58,8 +76,8 @@
 				</div>
 			</div>
 			<div class="buttons">
-				<SoftButton v-tooltip:bottom="t.bonus_point" icon="thumb_up" class="button-like" @click="counts.rating++" />
-				<SoftButton v-tooltip:bottom="t.minus_point" icon="thumb_down" class="button-dislike" @click="counts.rating--" />
+				<SoftButton v-tooltip:bottom="t.bonus_point" icon="thumb_up" class="button-like" @click="Upvote()" />
```
Comment (sidd) [inline components/Player/PlayerVideo/PlayerVideoPanel/PlayerVideoPanel.vue]: no need to call the function, here and below
```suggestion
				<SoftButton v-tooltip:bottom="t.bonus_point" icon="thumb_up" class="button-like" @click="Upvote" />
```

File: packages/kirakira-backend/DefaultApi.md [hunk 4]
```diff
@@ -340,6 +342,60 @@ No authorization required
...
 
+# **recommendations**
+> Array<Videos200ResponseVideosInner> recommendations()
+
+
+### Example
+
+
+```typescript
+import {  } from '';
```
Comment (sidd) [inline packages/kirakira-backend/DefaultApi.md:353]: ?.?

File: packages/kirakira-backend/DefaultApi.md [hunk 5]
```diff
@@ -340,6 +342,60 @@ No authorization required
...
+import * as fs from 'fs';
+
+const configuration = .createConfiguration();
```
Comment (sidd) [inline packages/kirakira-backend/DefaultApi.md:356]: seems  you need to specify the export and package name somewhere in the openapi config

File: pages/videos/kv[kvid].vue [hunk 6]
```diff
@@ -26,13 +27,29 @@
 		}).catch(error => console.error(error));
 	}, { immediate: true });
 
+	// Fetch recommendations
+	watch(() => kvid, () => {
```
Comment (sidd) [inline pages/videos/kv[kvid].vue]: nit: use async/await. same for other usages of `.then`

```typescript
watch(() => kvid, async () => {
	const api = useApi();
	try {
		const newRecommendations = await api.recommendations(kvid);
		recommmendations.value = newRecommendations;
	} catch (error) {
		console.error(error);
	}
}, { immediate: true });
```

File: pages/videos/kv[kvid].vue [hunk 7]
```diff
@@ -26,13 +27,29 @@
...
 	<div class="container">
-		<PlayerVideo v-if="videoSource !== undefined" :src="videoSource" />
-		<div class="under-player">
+		<PlayerVideo
+			v-if="videoSource !== undefined"
+			:id="videoDetails?.videoID ?? 0"
+			:src="videoSource"
+			:rating="videoDetails?.rating ?? 0"
+		/>
+		<div
```
Comment (sidd) [inline pages/videos/kv[kvid].vue]: it looks like the formatting here got rekt

File: pages/videos/kv[kvid].vue [hunk 8]
```diff
@@ -79,6 +102,14 @@
 		}
 	}
 
+	#recheader {
```
Comment (sidd) [inline pages/videos/kv[kvid].vue]: avoid using IDs unless you need to. vue scopes them, but it makes fighting for specificity harder

**Review by sidd — COMMENTED** [review]
random small stuff

File: packages/kirakira-backend/DefaultApi.md (hunk 4 above)
Comment (horahoradev) [reply packages/kirakira-backend/DefaultApi.md:353]: lol wtf happened there

Comment (horahoradev) [reply packages/kirakira-backend/DefaultApi.md:353]: oh but that's generated code, I didn't write it. IDK TBH

File: packages/kirakira-backend/DefaultApi.md (hunk 5 above)
Comment (horahoradev) [reply packages/kirakira-backend/DefaultApi.md:356]: alright let's try it...

Comment (horahoradev) [reply packages/kirakira-backend/DefaultApi.md:356]: nvm I don't see what the issue is, I'm already specifying npmName. To what extent does it matter?

File: pages/videos/kv[kvid].vue (hunk 6 above)
Comment (horahoradev) [reply pages/videos/kv[kvid].vue]: .then is a bad practice? I should use await?

Comment (horahoradev) [reply pages/videos/kv[kvid].vue]: oh i see, you wrote it the way i should write it

File: packages/kirakira-backend/DefaultApi.md (hunk 5 above)
Comment (sidd) [reply packages/kirakira-backend/DefaultApi.md:356]: not much, just makes the docs look ugly and not copy-pastable

**Comment by horahoradev** [comment]
should be poggers now

**Review by Aira-Sakuranomiya — APPROVED** [review]

**Pull request merged**